docs/modules/generate.md

```diff
@@ -0,0 +1,163 @@
+# 🏗️ Generate
+
+**`#bs.generate:help`**
+
+Generate diverse terrains and landscapes, incorporating natural features.
+
+```{image} /_imgs/modules/generate-light.png
+:align: center
+:class: only-light
+```
+
+```{image} /_imgs/modules/generate-dark.png
+:align: center
+:class: only-dark
+```
+
+```{epigraph}
+"Nature's beauty is a reflection of the harmony of numbers and patterns."
+
+-- Anonymous
+```
+
+---
+
+## 🔧 Functions
+
+You can find below all functions available in this module.
+
+---
+
+### Shape 2D
+
+:::::{tab-set}
+::::{tab-item} Shape 2D
+
+```{function} #bs.generate:shape_2d
+
+Generates a shape in 2D space at the specified origin, with the callback executed at each step, allowing the user to handle the X and Y coordinates directly.
+
+:Inputs:
+  **Execution `at <entity>` or `positioned <x> <y> <z> rotated <rot>`**: Origin of the shape.
+
+  **Function macro**:
+  :::{treeview}
+  - {nbt}`compound` Arguments
+    - {nbt}`compound` **run**: Callback to run at each step, responsible for handling the coordinates.
+    - {nbt}`compound` **width**: Width of the shape to generate.
+    - {nbt}`compound` **height**: Height of the shape to generate.
+    - {nbt}`compound` **with**: Shape settings.
+      - {nbt}`string` **direction**: Plane used to generate the shape (`xz`, `xy`, or `zy`, default: `xz`).
+      - {nbt}`int` **spacing**: Distance between blocks in the generated shape (default: 1).
+      - {nbt}`int` **limit**: Limit how many steps are executed in a single tick (default: 256).
+  :::
+
+:Outputs:
+  **State**: The callback gets executed at the appropriate coordinates in the world.
+  :::{treeview}
+  - For each step, the following scores can be used:
+    - **`$generate.x bs.data`**: The X coordinate of the current step.
+    - **`$generate.y bs.data`**: The Y coordinate of the current step.
```

Comment:
For me, should be put with the "run" parameter and not here

Reply:
Yeah i was'nt happy with this either. Good idea, i'll try to squeeze this below the run param.